adapters/nextmillennium/nextmillennium.go

```diff
@@ -99,12 +99,35 @@ func (adapter *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidReques
 }
 
 func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium) *NextMillenniumBidRequest {
+	placementID := params.PlacementID
 	bidRequest := NextMillenniumBidRequest{
 		ID:   prebidBidRequest.ID,
 		Test: uint8(prebidBidRequest.Test),
 	}
-	bidRequest.Ext.Prebid.StoredRequest.ID = params.PlacementID
 
+	if params.GroupID != "" {
+		domain := ""
+		size := ""
+
+		if prebidBidRequest.Site != nil {
+			domain = prebidBidRequest.Site.Domain
+		}
+		if prebidBidRequest.App != nil {
+			domain = prebidBidRequest.App.Domain
+		}
+
+		if banner := prebidBidRequest.Imp[0].Banner; banner != nil {
+			if len(banner.Format) > 0 {
+				size = fmt.Sprintf("%dx%d", banner.Format[0].W, banner.Format[0].H)
+			} else if banner.W != nil || banner.H != nil {
```

Comment:
To test this branch, please add the following json test cases:

1 - Banner impression without a format and with W and H defined:
```
"banner": { },
```

2 - Banner impression without a format and just one W or H defined:
```
"banner": {
  "w": 320
},
```

3 - Banner impression without a format and both W and H defined:
```
"banner": {
  "w": 320,
  "h": 250
},
```

Reply:
@SyntaxNode Hello!
All the changes you requested have been made